server/src/main/java/org/opensearch/index/fielddata/IndexNumericFieldData.java

```diff
@@ -96,6 +99,12 @@ public final boolean isFloatingPoint() {
         public final ValuesSourceType getValuesSourceType() {
             return valuesSourceType;
         }
+
+        public void setSortFieldType(SortField.Type type) {
```

Comment:
May be we could use constructor for that (with `usePointBasedOptimization` argument) instead of inplace modification?

Reply:
The numeric type is already initialized here. We are chaning that during run time to make it backward compatible.
To change the NumericType upfront thrugh constructor, that change will be lot of overlapping chnage where this case is not applicable at all...

Reply:
I understand that the numeric type is already initialized here. The solution clearly violates the `NumericType` invariants and I would argue that should not be the case. We have to look for a better ways to propagate `usePointBasedOptimization` change.

Reply:
Damn !! My bad.
Yeah, I reaized I modified enum and that is the reason it was giving type mimatch during search as well hence I had to disable optimization.

But that is absolutely not needed. I am not modifying enum now but just up-casting SortField.Type works fine during IndexSortField creation for IndexWriter.

Reply:
> But that is absolutely not needed. I am not modifying enum now but just up-casting SortField.Type works fine during IndexSortField creation for IndexWriter.

My apologies, for some reason your reply was not showing up when I posted the suggestion, please feel free to discard it if simpler option works, thanks!

Reply:
No problem.
Yes, I think now that `usePointBasedOptimization` is not required, lets not touch the existing NumericTypes and `sortField` methods. Everything is contained within `wideSortField` method for this backward compatibility handling.